### DISCUSSION TITLE
image workflow concern

### DISCUSSION THREAD

**Discussion by curryprogrammer** [dokku/dokku#4467] [Q&A] [answered]
how can i accomplish workflow described here:

https://dokku.com/docs/deployment/methods/images/#image-workflows

without using tags plugin? (which is marked as deprecated and is scheduled to be removed in future versions)

**Comment by josegonzalez** [comment, marked as answer]
Probably something like

```
dokku apps:create test
dokku git:from-image test demo-repo/some-image:v12
```

**Reply by jasonlimantoro** [reply]
What if the docker image is in a private registry, how to authenticate?

**Reply by jasonlimantoro** [reply]
Right now this workaround works (for private registry).

## Setup Docker Auth
```
docker login $DOCKER_IMAGE -u $USER -p $TOKEN
```
> PS: I'm using GitLab, so replace the above variables appropriately. The token is an access token created in the Gitlab UI.

## Initial Dokku Setup
```
dokku apps:create $APP
dokku git:from-image $APP $DOCKER_IMAGE
```

## Update Dokku App
1. Push your image to the registry (for example, from CI/CD or from a local machine using `docker build` and `docker push`).
2. Update the dokku App
```
docker pull $DOCKER_IMAGE
dokku ps:rebuild $APP
```

Not sure if this is documented or the best way to do it.

**Reply by josegonzalez** [reply]
Yeah a `docker login` is necessary. I think that command will fall under the registry plugin.

**Reply by josegonzalez** [reply]
You don't need a `ps:rebuild` unless the image has the exact same tag (in which case you should use an image digest instead of the tag name). You should avoid using the same image tag for each deploy.
